auth/credentials/detect.go

```diff
@@ -250,3 +253,31 @@ func clientCredConfigFromJSON(b []byte, opts *DetectOptions) *auth.Options3LO {
 		AuthStyle: auth.StyleInParams,
 	}
 }
+
+// GetGoogleMtlsTokenURL returns the mTLS token exchange URL
+func GetGoogleMtlsTokenURL() string {
+	return googleMtlsTokenURL
+}
+
+// CustomHTTPClient constructs an HTTPClient using the provided tlsConfig, to support mTLS.
+func CustomHTTPClient(tlsConfig *tls.Config) *http.Client {
+	trans := baseTransport()
+	trans.TLSClientConfig = tlsConfig
+	return &http.Client{Transport: trans}
+}
+
+func baseTransport() *http.Transport {
```

Comment:
I feel like we have this code somewhere else as well. Could it be referenced instead.

Reply:
This is a port of the logic from the old library - it provides a fully configured based HTTP client to use with OAuth2 token exchange over mTLS - we don't have an existing implementation that fits the bill, but could consider relocating the impl in the future.